framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1211,7 +1211,11 @@ private AnnotatedTypeMirror fromExpression(ExpressionTree tree) {
 
         AnnotatedTypeMirror result = TypeFromTree.fromExpression(this, tree);
 
-        if (shouldCache) {
+        if (shouldCache && tree.getKind() != Tree.Kind.NEW_CLASS) {
+            // Don't cache types of NEW operator
```

Comment:
How about:

```
// Don't cache the type of object creations, because incorrect
// annotations would be cached during dataflow analysis.
// See Issue #602.
```